test/fluree/db/reasoner/owl_datalog_test.clj

```diff
@@ -0,0 +1,401 @@
+                                     "@type"        ["ex:ManufacturedItem" "ex:SubstanceDefinition"]
+                                     "ex:conformsTo" {"@id"   "ex:aspirin-formulation"
+                                                      "@type" "ex:Formulation"}}}
+
+          db-with-data @(fluree/update db-with-ontology instance-data)
+
+          ;; Apply owl-datalog reasoning
+          db-reasoned @(fluree/reason db-with-data :owl-datalog)]
+
+      (testing "Instance with all intersection components should be inferred as DrugProduct"
+        (is (= #{"ex:DrugProduct" "ex:ManufacturedItem" "ex:SubstanceDefinition"}
+               (set @(fluree/query db-reasoned
+                                   {:context {"ex" "http://example.org/"}
+                                    :select  "?type"
+                                    :where   {"@id"   "ex:aspirin-tablet"
+                                              "@type" "?type"}})))
+            "ex:aspirin-tablet should be inferred as ex:DrugProduct")))))
+
+(deftest ^:integration someValuesFrom-in-equivalentClass-test
+  (testing "owl:someValuesFrom in owl:equivalentClass definition"
+    ;; Pattern: Class ≡ ∃property.ValueClass
+    ;; If x has property y and y is of type ValueClass, then x is of type Class
+    (let [conn (test-utils/create-conn)
+          db @(fluree/create conn "reasoner/somevalues-equiv" nil)
+
+          ontology {"@context" {"ex"   "http://example.org/"
+                                "owl"  "http://www.w3.org/2002/07/owl#"
+                                "rdfs" "http://www.w3.org/2000/01/rdf-schema#"}
+                    "insert"   [{"@id"                  "ex:Parent"
+                                 "@type"                "owl:Class"
+                                 "owl:equivalentClass"  {"@type"              "owl:Restriction"
+                                                         "owl:onProperty"     {"@id" "ex:hasChild"}
+                                                         "owl:someValuesFrom" {"@id" "ex:Person"}}}
+                                {"@id"   "ex:Person"
+                                 "@type" "owl:Class"}
+                                {"@id"   "ex:hasChild"
+                                 "@type" "owl:ObjectProperty"}]}
+
+          db-with-ontology @(fluree/update db ontology)
+
+          instance-data {"@context" {"ex" "http://example.org/"}
+                         "insert"   [{"@id"        "ex:john"
+                                      "ex:hasChild" {"@id" "ex:mary"}}
+                                     {"@id"   "ex:mary"
+                                      "@type" "ex:Person"}]}
+
+          db-with-data @(fluree/update db-with-ontology instance-data)
+          db-reasoned @(fluree/reason db-with-data :owl-datalog)]
+
+      (testing "Individual with property pointing to correct type should be inferred as Parent"
+        (is (contains? (set @(fluree/query db-reasoned
```

Comment:
Same thing here, there should only be a single result type, so no need to put it in a set.